core/src/poh_service.rs

```diff
@@ -254,24 +259,38 @@ impl PohService {
                     timing.num_hashes += hashes_per_batch;
                     let mut hash_time = Measure::start("hash");
                     let should_tick = poh_l.hash(hashes_per_batch);
+                    let ideal_time = poh_l.target_poh_time(target_ns_per_tick);
                     hash_time.stop();
                     timing.total_hash_time_ns += hash_time.as_ns();
                     if should_tick {
                         // nothing else can be done. tick required.
                         return true;
                     }
                     // check to see if a record request has been sent
-                    let get_again = record_receiver.try_recv();
-                    match get_again {
-                        Ok(record) => {
+                    if let Ok(record) = record_receiver.try_recv() {
+                        // remember the record we just received as the next record to occur
+                        *next_record = Some(record);
```

Comment:
Separate thought: Because `PoHRecorder::record()` can tick(), without incurring the spinwait/throttle, if we had enough records, would that entirely circumvent the throttle/waits? Maybe this is highly unlikely unless we have a bunch of banking threads...

Reply:
we still throttle in pohrecorder::tick. So, if pohrecorder::record calls pohrecorder::tick(), we still throttle. The goal of this is to reduce the useless work spinning in pohrecorder::tick and check for records while waiting for time to elapse in pohservice.

Reply:
ah yup you're right, that `spin_loop` is in the `tick()` function itself, not inside the body of  PoH service.